xblock/fields.py

```diff
@@ -779,6 +779,18 @@ def from_json(self, value):
 
     enforce_type = from_json
 
+    def to_string(self, value):
+        """
+        In python3, json.dumps() cannot sort keys of different types, 
+        so preconvert None to 'null'.
+        """
+        self.enforce_type(value)
+        if isinstance(value, dict) and None in value:
+            value = value.copy()
+            value['null'] = value[None]
+            del value[None]
+        return super(Dict, self).to_string(value)
```

Comment:
Hrm... we don't explicitly disallow numeric keys on the python side, so for completeness, we probably should be checking for numeric keys. This is all making dropping the `sort_keys` parameter sound better to me, just from the perspective of limiting the amount of preprocessing we have to do over and above what the json module already does.